server/src/main/java/org/opensearch/action/pagination/ShardPaginationStrategy.java

```diff
@@ -54,39 +64,31 @@ public ShardPaginationStrategy(PageParams pageParams, ClusterState clusterState)
         );
     }
 
-    private static List<IndexMetadata> getEligibleIndices(
-        ClusterState clusterState,
+    private static Predicate<IndexMetadata> getMetadataFilter(
         String sortOrder,
         String lastIndexName,
-        Long lastIndexCreationTime
+        Long lastIndexCreationTime,
+        IndicesOptions indicesOptions
     ) {
         if (Objects.isNull(lastIndexName) || Objects.isNull(lastIndexCreationTime)) {
-            return PaginationStrategy.getSortedIndexMetadata(
-                clusterState,
-                PageParams.PARAM_ASC_SORT_VALUE.equals(sortOrder) ? ASC_COMPARATOR : DESC_COMPARATOR
-            );
-        } else {
-            return PaginationStrategy.getSortedIndexMetadata(
-                clusterState,
-                getMetadataFilter(sortOrder, lastIndexName, lastIndexCreationTime),
-                PageParams.PARAM_ASC_SORT_VALUE.equals(sortOrder) ? ASC_COMPARATOR : DESC_COMPARATOR
-            );
-        }
-    }
-
-    private static Predicate<IndexMetadata> getMetadataFilter(String sortOrder, String lastIndexName, Long lastIndexCreationTime) {
-        if (Objects.isNull(lastIndexName) || Objects.isNull(lastIndexCreationTime)) {
-            return indexMetadata -> true;
+            return indexStateFilter(indicesOptions);
         }
         return indexNameFilter(lastIndexName).or(
             IndexPaginationStrategy.getIndexCreateTimeFilter(sortOrder, lastIndexName, lastIndexCreationTime)
-        );
+        ).and(indexStateFilter(indicesOptions));
     }
 
     private static Predicate<IndexMetadata> indexNameFilter(String lastIndexName) {
         return metadata -> metadata.getIndex().getName().equals(lastIndexName);
     }
 
+    private static Predicate<IndexMetadata> indexStateFilter(IndicesOptions indicesOptions) {
+        if (Objects.isNull(indicesOptions) || !indicesOptions.forbidClosedIndices()) {
```

Comment:
I think `Objects.isNull` exists to use as a predicate. The canonical way to do this is just `indicesOptions == null`

Reply:
Addressed in the revision.